### ISSUE TITLE
Media download button

### ISSUE THREAD

**Issue by KhazAkar** [hjiangsu/thunder#50]
**Is your feature request related to a problem? Please describe.**
Simple thing - I can see a funny meme, but cannot download it, or some video, or anything else.

**Describe the solution you'd like**
Simple download button when you tap on media to play it

**Describe alternatives you've considered**
Screenshot in case of images, nothing in case of gif/video

**Comment by MrAntonS** [comment]
Working on it

**Comment by MrAntonS** [comment]
So I made it work, but... I have no idea why it works without any permission required. I would need the main author to check the code and suggest changes XD

**Comment by hjiangsu** [comment]
Thanks! I'll assign you to this issue

**Comment by hjiangsu** [comment]
> So I made it work, but... I have no idea why it works without any permission required. I would need the main author to check the code and suggest changes XD

This could very well be a difference between the debug version and release versions allowing for certain permissions automatically. I found out when I released the first build of the app that the debug builds automatically adds in the INTERNET permission into the `AndroidManifest.xml` file

This is the particular discussion about this for context: https://github.com/hjiangsu/thunder/issues/2#issuecomment-1595341124

```xml
<uses-permission android:name="android.permission.INTERNET" />
```

**Comment by MrAntonS** [comment]
I see. Yes i added Storage Permission. I guess that's why it didn't ask for anything.. I'll have to test it next week. Check out my code and give any suggestions! 😄

**Comment by hjiangsu** [comment]
Thanks! I'll take a look at the PR when I have the chance :D

**Comment by hjiangsu** [comment]
Closing as this should be fixed in v0.2.1+10!
